src/native/libs/System.Net.Security.Native.Apple/pal_networkframework.m

```diff
@@ -0,0 +1,433 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_networkframework.h"
+#include <Foundation/Foundation.h>
+
+static WriteCallback _writeFunc;
+static ReadCallback _readFunc;
+static StatusUpdateCallback _statusFunc;
+static nw_protocol_definition_t _framerDefinition;
+static nw_protocol_definition_t _tlsDefinition;
+static dispatch_queue_t _tlsQueue;
+static dispatch_queue_t _inputQueue;
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunguarded-availability-new"
+
+PALEXPORT nw_connection_t AppleNetNative_NwCreateContext(int32_t isServer)
+{
+    if (isServer != 0)  // only client supported at this point.
+        return NULL;
+
+    nw_parameters_t nw_parameters = nw_parameters_create_secure_udp(NW_PARAMETERS_DISABLE_PROTOCOL, NW_PARAMETERS_DEFAULT_CONFIGURATION);
+    nw_endpoint_t nw_endpoint = nw_endpoint_create_host("127.0.0.1", "42");
+
+    nw_connection_t connection = nw_connection_create(nw_endpoint, nw_parameters);
+
+    return connection;
+}
+
+// This writes encrypted TLS frames to the safe handle. It is executed on NW Thread pool
+static nw_framer_output_handler_t framer_output_handler = ^(nw_framer_t framer, nw_framer_message_t message, size_t message_length, bool is_complete)
+{
+    nw_protocol_options_t framer_options;
+
+    if (__builtin_available(macOS 12.3, iOS 15.4, tvOS 15.4, watchOS 2.0, *))
+    {
+        framer_options = nw_framer_copy_options(framer);
+
+        NSNumber* num = nw_framer_options_copy_object_value(framer_options, "GCHANDLE");
+        assert(num != NULL);
+
+        void * ptr;
+        [num getValue:&ptr];
+        size_t size = message_length;
+
+        nw_framer_parse_output(framer, 1, message_length, NULL, ^size_t(uint8_t *buffer, size_t buffer_length, bool is_complete2) {
+            size_t length = buffer_length;
+            (_writeFunc)(ptr, buffer, length);
+            (void)is_complete2;
+            (void)message;
+            return buffer_length;
+        });
+    }
+    else
+    {
+        assert(0);
```

Comment:
What happens here in a release build when the assert evaporates? Will the consumer behave in a desired fashion?